File: explicaciones/programacion/py_vs_rb/A2/program2.py
import sys
import math

x1 = int(sys.argv[1])
x2 = int(sys.argv[3])
x3 = int(sys.argv[5])
y1 = int(sys.argv[2])
y2 = int(sys.argv[4])
y3 = int(sys.argv[6])

d1 = math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)
d2 = math.sqrt((x1 - x3) ** 2 + (y1 - y3) ** 2)

if d2 > d1:
    print('El punto más cercano a','(',x1,',',y1,
    ') es (',x2,',',y2,') y está a una distancia',d1)
elif d2 < d1:
    print('El punto más cercano a','(',x1,',',y1,
    ') es (',x3,',',y3,') y está a una distancia',d2)

File: explicaciones/ruby/py_vs_rb/a2/program2.rb
#!/usr/bin/ruby

x1 = ARGV[0].to_i
x2 = ARGV[2].to_i
x3 = ARGV[4].to_i
y1 = ARGV[1].to_i
y2 = ARGV[3].to_i
y3 = ARGV[5].to_i

d1 = Math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)
d2 = Math.sqrt((x1 - x3) ** 2 + (y1 - y3) ** 2)

if d2 > d1
  puts "El punto más cercano a (#{x1}, #{y1}) es (#{x2}, #{y2}) y está a una distancia #{d1}"
elsif d2 < d1
  puts "El punto más cercano a (#{x1}, #{y1}) es (#{x3}, #{y3}) y está a una distancia #{d2}"
end
